### PR TITLE
Direction change + "Home" & "End" keys

### PR DESCRIPTION
**Pull request by StevenMeyer**
Up/down arrows used for vertical roving. Home and End keys rove directly to start and end of tab stops respectively.

### CODE REVIEW

File: src/Provider.tsx [hunk 1]
```diff
@@ -100,6 +127,20 @@ function reducer(state: State, action: Action): State {
         tabStops
       };
     }
+    case ActionTypes.TAB_TO_FIRST:
```
Comment (olijyat) [inline src/Provider.tsx:130]: This supposed to be empty?

Comment (StevenMeyer) [reply src/Provider.tsx:130]: It's not empty. `TAB_TO_FIRST` and `TAB_TO_LAST`  use the same block. I've copied the switch-fallthrough style from `TAB_TO_PREVIOUS`/`TAB_TO_NEXT`.

File: src/Provider.tsx [hunk 2]
```diff
@@ -100,6 +127,20 @@ function reducer(state: State, action: Action): State {
...
+    case ActionTypes.TAB_TO_LAST: {
+      const id = action.payload.id;
```
Comment (olijyat) [inline src/Provider.tsx]: const { id } = action.payload;

Comment (StevenMeyer) [reply src/Provider.tsx]: This code is gone

**Pull request merged**